### DISCUSSION TITLE
Clarification on Edge Weights in the Adjacency Matrix

### DISCUSSION THREAD

**Discussion by Firedrops** [medmaca/PGCNA#6] [General]
Hi!

So I've got PGCNA fully running on my datasets... but I haven't found a clear explanation of what edge weights represent.

I'm getting both -ve and +ve values, seems bound within -1 to 1.

By reading the paper and some intuition, I think:
1. Retained edges have already all passed the P-value <0.05 threshold. So the method is _fairly_ certain that _a_ correlation exists.
2. Edge weights is the strength of correlation, with a multiplier... so if Gene X to Gene Y has an edge weight of 0.9, then every 100% increase in expression of Gene X *should* correspond to a 90% increase of Gene Y.
3. And -ve is simply reverse correlation... so if Gene X to Gene Y has an edge weight of -3, then every 100% increase in expression of Gene X *should* correspond to a 90% **_decrease_** of Gene Y.

Please let me know whether my interpretation is correct?

**Comment by medmaca** [comment]
Hi,

Glad you've managed to get everything running.  

Are you referring to the values in the GEPHI/*_Edges.tsv files?  These are the edge correlations (either Spearman or Pearson depending on option selected).  They can range from -1 to 1.

However, I've not seen any negative correlations making it through to the _Edges.tsv file in any of the data I've looked at.

A few quick questions:

1. Is this a single dataset or multiple?
2. Is it bulk RNA/microarray?
3. Have you filtered the data in any way before analysing with PGCNA?

**Comment by Firedrops** [comment]
Oh, I was looking at the `.h5` files in `CORR_MATRIX_GMF0`, since I was not using GEPHI. Are they reporting different things?
1. Multiple datasets, but I was running them individually (only 1 entry in #FileInfo, each), so probably behaves as single dataset
2. It's a single-cell RNA dataset
3. Yes, I have gone through a standard filtration, and then isolated only the "Differentially Expressed" genes, which are generally consistently highly expressed, so sparsity _might_ not be a major concern.

**Comment by medmaca** [comment]
OK, so presuming that you're **not** using the `*NoEdgeReduction.h5` then the `.h5` file should contain the same information as the `GEPHI/*Edges.tsv` file.  Though the latter might be quicker to parse work with (depending on what you want to accomplish).

However:

1. PGCNA wasn't designed to work with single-cell RNA seq data... and thus you might get odd results with sparse matrices, or massively reduce gene numbers.
2. If you are going to use PGCNA for this data, I suggest using the Pearson correlation over the default Spearman.
3. The fact you're getting negative correlations suggests to me that you've got some genes that have no +ve correlation partners... which indicates a highly reduced number of genes, these are basically orphan genes.
4. I would suggest removing all negative correlation edges.  But again it really depends what you're using the data for moving forward?
5. If you are indeed looking at the `*NoEdgeReduction.h5` files then this will contain all the correlations (+ve/-ve) before edge reduction.

**Comment by Firedrops** [comment]
Yup, your observations are pretty much spot on. I did use Pearson, yes, and only on small sets of genes (only ~100-1000 as opposed to typically ~10000 in full datasets). 

But what do the -ve correlation edges mean? I may need to justify it to my peers for removing them.

**Comment by medmaca** [comment]
Just to be clear, can you clarify that you're definitely **not** looking at the `NoEdgeReduction.h5` files?

The -ve correlations mean that the two genes are anticorrelated (when one goes up the other goes down).  I'm surprised that for some genes (even with such a sparse matrix) that you're not discovering any +ve correlations (however weak).

For the genes in question do they have any +ve edges?

**Comment by Firedrops** [comment]
Ah sorry missed that part, the ones I'm looking at are `*RetainF1.0.h5` files, no `NoEdgeReduction` anywhere.

I've checked through a few datasets, it appears that most genes occur the exact same number of times in the resulting matrix, and it varies with how large of an input I fed to PGCNA2.

For example, on one dataset with 154 genes, in the resulting edge list, every gene appears ~153 times, but this is before consolidating the "directivity", so every edge is double-counted. My parameters were `-f 1 --usePearson --noLeidenalg`

Could you please also clarify what the numerical value of the edge weight represents? Is it the "confidence" in the correlation? Or the "amplification" of correlation? Or something else?

**Comment by medmaca** [comment]
The edge weight here is simply the correlation for each pair of genes.

It seems to me that the way you're currently analysing the data you might be better off just using the numpy Pearson correlation https://numpy.org/doc/stable/reference/generated/numpy.corrcoef.html method on your expression data.

However, again I'd stress that you should probably be looking at scRNA-seq tools (e.g. https://satijalab.org/seurat/) for analysing your data.
